packages/firestore/test/unit/local/indexeddb_persistence.test.ts

```diff
@@ -602,6 +586,66 @@ describe('IndexedDbSchema: createOrUpgradeDb', () => {
       });
     });
   });
+
+  const readManualVersion = (db: IDBDatabase): Promise<number> => {
+    const sdb = new SimpleDb(db);
+    return sdb.runTransaction('readonly', ['versionGlobal'], txn => {
+      const store = txn.store<string, number>('versionGlobal');
+      return store.get('versionGlobalKey').next(maybeVersion => {
+        expect(maybeVersion).to.not.be.undefined;
+        return maybeVersion!;
+      });
+    });
+  };
+
+  it('can save its own version number', async () => {
+    await withDb(8, async db => {
+      expect(db.version).to.equal(8);
+      const version = await readManualVersion(db);
+      expect(version).to.equal(8);
+    });
+  });
+
+  it('does not delete object stores', async () => {
+    // Since we intend on allowing downgrades, we cannot delete previously
+    // existing object stores. Verify that after each individual migration, we
+    // still have the object stores that we started with.
+    let stores: string[] = [];
+    for (let version = 1; version <= SCHEMA_VERSION; version++) {
+      await withDb(version, async db => {
+        const thisVersionStores = db.objectStoreNames;
+        for (const store of stores) {
+          expect(thisVersionStores.contains(store)).to.be.true;
+        }
+        // Reload the existing stores to check the next version.
+        stores = [];
+        for (let i = 0; i < thisVersionStores.length; i++) {
+          stores.push(thisVersionStores[i]);
+        }
+      });
+    }
+  });
+
+  it('can upgrade, then downgrade to earliest safe point, then upgrade again', async () => {
+    // Upgrade to latest version
+    await withDb(SCHEMA_VERSION, async db => {
+      expect(db.version).to.equal(FIRST_MANUAL_SCHEMA_VERSION);
+      const version = await readManualVersion(db);
+      expect(version).to.equal(SCHEMA_VERSION);
+    });
+    // downgrade to first manual version
+    await withDb(FIRST_MANUAL_SCHEMA_VERSION, async db => {
+      expect(db.version).to.equal(FIRST_MANUAL_SCHEMA_VERSION);
+      const version = await readManualVersion(db);
+      expect(version).to.equal(FIRST_MANUAL_SCHEMA_VERSION);
+    });
+    // upgrade to latest version again
+    await withDb(SCHEMA_VERSION, async db => {
```

Comment:
You've probably already done this, but maybe set a breakpoint here and on the dummy migration, and just make sure it gets called again.... since the test wouldn't notice if we updated the version number but didn't re-run the migrations (I think).